### PR TITLE
Profiling code for dendrites. 

### PR DESCRIPTION
**Pull request by akashvelu**
Code to profile dendrites with torch profiler and/or notebooks for wall time.

### CODE REVIEW

File: projects/gsc/experiments/sampled_kwinners/kwinner_functions.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import torch
```
Comment (lucasosouza) [inline projects/gsc/experiments/sampled_kwinners/kwinner_functions.py:1]: Is this gsc/kwinners included intentionally? It doesn't directly tied to profiling dendrites.

File: projects/gsc/experiments/sampled_kwinners/sampled_le_sparse_net.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
```
Comment (lucasosouza) [inline projects/gsc/experiments/sampled_kwinners/sampled_le_sparse_net.py:1]: Same as before, was this included intentionally?

File: nupic/research/frameworks/dendrites/routing/utils.py [hunk 3]
```diff
@@ -34,6 +33,10 @@
 ]
 
 
+def generate_context_integers(num_contexts):
```
Comment (lucasosouza) [inline nupic/research/frameworks/dendrites/routing/utils.py:36]: I understand you are building another PR with the code optimizations - if this is related to the optimizations, maybe we should move it there, so this PR is specific to profiling

**Review by lucasosouza — COMMENTED** [review]
Left some comments.

File: projects/gsc/experiments/sampled_kwinners/kwinner_functions.py (hunk 1 above)
Comment (akashvelu) [reply projects/gsc/experiments/sampled_kwinners/kwinner_functions.py:1]: Yeah, this was an accident. Will update the PR to get rid of the other stuff.

File: projects/gsc/experiments/sampled_kwinners/sampled_le_sparse_net.py (hunk 2 above)
Comment (akashvelu) [reply projects/gsc/experiments/sampled_kwinners/sampled_le_sparse_net.py:1]: Removed the changes from the other files.

**Pull request merged**